### PR TITLE
Complete SOC support for STM32G4 series

### PR DESCRIPTION
**Pull request by gudnimg**
This PR completes the SOC support for STMicroelectronics STM32G4 series.

This will reduce work to add new boards with these MCUs.

MCUs included in this PR:

* STM32G4x1 https://www.st.com/en/microcontrollers-microprocessors/stm32g4x1.html
    * STM32G4A1xE (512K / 112K)
    * STM32G491xE (512K / 112K)
    * STM32G491xC (256K / 112K)
    * STM32G441xB (128K / 32K)
    * [Already supported] STM32G431xB (128K / 32K)
    * STM32G431x8 (64K / 32K)
    * STM32G431x6 (32K / 32K)

* STM32G4x3 https://www.st.com/en/microcontrollers-microprocessors/stm32g4x3.html
    * STM32G483xE (512K / 128K)
    * STM32G473xE (512K / 128K)
    * STM32G473xC (256K / 128K)
    * STM32G473xB (128K / 128k)

* STM32G4x4 https://www.st.com/en/microcontrollers-microprocessors/stm32g4x4.html
    * STM32G484xE (512K / 128K)
    * [Already supported]  STM32G474xE (512K / 128K)
    * [Already supported]  STM32G474xC (256K / 128K)
    * [Already supported]  STM32G474xB (128K / 128K)

### CODE REVIEW

File: dts/arm/st/g4/stm32g431X8.dtsi [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2021 Guðni Már Gilbert
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <st/g4/stm32g431.dtsi>
```
Comment (gmarull) [inline dts/arm/st/g4/stm32g431X8.dtsi:8]: nit: this change should likely belong to 71170107f470681ce23df61e2a0a8843f0a1c8a2

File: dts/arm/st/g4/stm32g484.dtsi [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2021 Guðni Már Gilbert
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/g4/stm32g4.dtsi>
```
Comment (gmarull) [inline dts/arm/st/g4/stm32g484.dtsi:7]: if this file is empty, `#include <st/g4/stm32g4.dtsi>` may be used instead

File: dts/arm/st/g4/stm32g431X8.dtsi (hunk 1 above)
Comment (GunZi200) [reply dts/arm/st/g4/stm32g431X8.dtsi:8]: Good catch! 💪you’re right.

File: dts/arm/st/g4/stm32g484.dtsi (hunk 2 above)
Comment (GunZi200) [reply dts/arm/st/g4/stm32g484.dtsi:7]: I forgot to add the DMA nodes similar to the other files. Will fix that quickly.

File: dts/arm/st/g4/stm32g441.dtsi [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2021 Guðni Már Gilbert
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/g4/stm32g431.dtsi>
```
Comment (gmarull) [inline dts/arm/st/g4/stm32g441.dtsi:7]: this file is also empty, is DMA stuff missing ?

Comment (GunZi200) [reply dts/arm/st/g4/stm32g441.dtsi:7]: No it is included from `st/g4/stm32g431.dtsi` since it is the same.

Comment (gmarull) [reply dts/arm/st/g4/stm32g441.dtsi:7]: so I'd just include 431 directly

Comment (gmarull) [reply dts/arm/st/g4/stm32g441.dtsi:7]: nevermind, it's ok, other soc seem to do that.

**Review by gmarull — APPROVED** [review]
Thanks for this addition!

**Review by erwango — APPROVED** [review]

**Pull request merged**